crates/biome_js_analyze/src/lint/nursery/use_import_extensions.rs

```diff
@@ -64,6 +70,25 @@ declare_rule! {
     /// ```js
     /// require("./foo.js");
     /// ```
+    ///
+    /// ### Options
+    ///
+    /// Use the options to specify correct import extensions for your project based on the linted
+    /// file extension. These mappings will override the default guesses from the rule.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "suggestedExtensions": {
+    ///             "ts": {
+    ///                 "import": "js",
+    ///                 "component": "jsx"
+    ///             }
```

Comment:
I believe we should be more descriptive here. We should explain the example, and we should explain what `import` and how it affects the code action, and we should explain what `component` is and how `component` affects the code action.

Usually, we follow this format:
1. Talk about the options
2. Explain the example
3. Code block of the example

___

To give you better feedback. I am reading this for the first time, and I still don't understand how these options work. For example `"ts"` as a key isn't explained. Is it an input? An output?

Reply:
I think we could add some rationale. We could inform users that it is a temporary workaround. Ideally we should query the extension of the imported file and choose an extension based on that.

Reply:
I have updated the description and tried to provide a deeper explanation for the reasoning and way to use that in a project. Do you mind taking another look?